client/network/src/discovery.rs

```diff
@@ -137,17 +137,9 @@ impl DiscoveryConfig {
 	}
 
 	/// Add discovery via Kademlia for the given protocol.
-	pub fn add_protocol(&mut self, p: ProtocolId) -> &mut Self {
-		// NB: If this protocol name derivation is changed, check if
-		// `DiscoveryBehaviour::new_handler` is still correct.
-		let proto_name = {
-			let mut v = vec![b'/'];
-			v.extend_from_slice(p.as_bytes());
-			v.extend_from_slice(b"/kad");
-			v
-		};
-
-		self.add_kademlia(p, proto_name);
+	pub fn add_protocol(&mut self, id: ProtocolId) -> &mut Self {
+		let name = protocol_name_from_protocol_id(&id);
+		self.add_kademlia(id, name);
```

Comment:
This change seems totally unnecessary and I do actually prefer the current version because you don't have to scroll down at the bottom of the file to see what happens.

Reply:
Within the 3 unit tests below I need to create a _protocol name_ from a _protocol id_. As the test is not testing this logic itself I was reluctant to replicate the logic within our unit tests.

With that in mind would you still prefer this logic to be inlined (and thus replicated in the unit tests below)?

Reply:
I see. In my mind we would ideally rework this a bit, but it can be left like that.